api/rpc/stat/stats.go

```diff
@@ -0,0 +1,17 @@
+package stat
```

Comment:
I don't think we need an interface for stat. I don't see us using a factory pattern to create instances from different implementations unless we also have an effective abstraction for the query interface. In any case, I think the extra effort right now is premature. I would just make the `stat` type public (`Stat`) in your implementation file and have `New()` return a new instance of it directly for now. It would be easy enough to return an interface instead in the future if that becomes necessary.